packages/foam-vscode/src/features/language-server-client/index.ts

```diff
@@ -0,0 +1,74 @@
+import * as path from "path";
+import { workspace, ExtensionContext } from "vscode";
+import { FoamFeature } from "../../types";
+
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+  TransportKind
+} from "vscode-languageclient";
+
+let client: LanguageClient;
+
+const feature: FoamFeature = {
+  activate: (context: ExtensionContext) => {
+    // @TODO figure out path for production deploys
+    let serverModule = context.asAbsolutePath(
+      path.join("..", "foam-language-server", "dist", "index.js")
+    );
+
+    // The debug options for the server
+    // --inspect=6009: runs the server in Node's Inspector mode so VS Code can attach to the server for debugging
+    let debugOptions = { execArgv: ["--nolazy", "--inspect=6009"] };
+
+    // If the extension is launched in debug mode then the debug server options are used
+    // Otherwise the run options are used
+    let serverOptions: ServerOptions = {
+      run: { module: serverModule, transport: TransportKind.ipc },
+      debug: {
+        module: serverModule,
+        transport: TransportKind.ipc,
+        options: debugOptions
+      }
+    };
+
+    // Options to control the language client
+    let clientOptions: LanguageClientOptions = {
+      // Register the server for plain markdown documents
+      documentSelector: [
+        { scheme: "file", language: "markdown" },
+        { scheme: "file", language: "mdx" }
+      ],
+      synchronize: {
+        fileEvents: workspace.createFileSystemWatcher(
+          "**/.vscode/settings.json"
```

Comment:
is this enough? would it also catch e.g. files being renamed/moved?

Reply:
This only watches the settings file for settings synchronisation. 

LSP has a [direct mechanism for giving updates to changed files in open buffers](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_synchronization) that doesn't require an explicit watcher. I'm not sure how we'll monitor background changes yet.